### PR TITLE
Conv1d

### PR DESCRIPTION
**Pull request by thecharlieblake**
Implementing a Conv1d function and module, building on #66 by @norikazu99.

### CODE REVIEW

File: unit_scaling/_modules.py [hunk 1]
```diff
@@ -181,6 +182,73 @@ def forward(self, input: Tensor) -> Tensor:
...
+            dtype,
+        )
+        self.kernel_size = kernel_size  # type:ignore[assignment]
+        self.stride = stride  # type:ignore[assignment]
+        self.padding = padding  # type:ignore[assignment]
+        self.dilation = dilation  # type:ignore[assignment]
+        self.constraint = constraint
+        self.weight = Parameter(self.weight.data, mup_type=weight_mup_type)
+        if self.bias is not None:
+            self.bias = Parameter(self.bias, mup_type="bias")
```
Comment (DouglasOrr) [inline unit_scaling/_modules.py:229]: (I notice this is in `Linear` too, so my bad) - perhaps this should be `self.bias.data` for consistency, although I suspect it doesn't matter.

File: unit_scaling/functional.py [hunk 2]
```diff
@@ -262,6 +262,49 @@ def linear_readout(
...
+@docstring_from(
+    F.conv1d,
+    short_description="Applies a **unit-scaled** 1D convolution.",
+    add_args=[
+        binary_constraint_docstring,
+        "scale_power ((float, float, float), optional): scaling power"
+        " for each of (output, grad(input), grad(weight|bias))",
+    ],
+)
+def conv1d(
```
Comment (DouglasOrr) [inline unit_scaling/functional.py:274]: LGTM, brill - thanks @thecharlieblake and @norikazu99 for this!

File: unit_scaling/functional.py [hunk 3]
```diff
@@ -262,6 +262,49 @@ def linear_readout(
...
+    stride: int = 1,
+    padding: int = 0,
+    dilation: int = 1,
+    groups: int = 1,
+    constraint: Optional[str] = "to_output_scale",
+    scale_power: Tuple[float, float, float] = (0.5, 0.5, 0.5),
+) -> Tensor:
+    fan_out, fan_in, kernel_size = weight.shape
+    seq_len = input.shape[-1]
+    out_size = (seq_len + 2 * padding - dilation * (kernel_size - 1) - 1) // stride + 1
```
Comment (DouglasOrr) [inline unit_scaling/functional.py:287]: (I was relieved to find this equation in the docs, as it's a headache!)

**Review by DouglasOrr — APPROVED** [review]
LGTM; just one thing which was more of a pre-existing issue. (Can be tidied up separately if wanted.)

File: unit_scaling/_modules.py [hunk 4]
```diff
@@ -181,6 +182,73 @@ def forward(self, input: Tensor) -> Tensor:
...
+    add_args=[binary_constraint_docstring],
+)
+class Conv1d(nn.Conv1d):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: int,
+        stride: int = 1,
+        padding: Union[str, int] = 0,
```
Comment (EIFY) [inline unit_scaling/_modules.py:199]: This implementation can't handle `str` padding parameter: It errors out if we pass in `padding='valid'` in `test_conv1d()` in `test_modules.py`:
```python
def test_conv1d() -> None:
    batch_size = 2**6
    d_in = 2**6 * 3
    d_out = 2**6 * 5
    kernel_size = 11
    seq_len = 2**6 * 7
    input = randn(batch_size, d_in, seq_len, requires_grad=True)
    model = Conv1d(d_in, d_out, kernel_size, padding='valid', bias=True)
    output = model(input)
```
```python
FAILED unit_scaling/tests/test_modules.py::test_conv1d - TypeError: unsupported operand type(s) for +: 'int' and 'str'
```
That said, `str` padding parameter seems to be a can of worms. Maybe just tighten it to only support `padding: int` for now?

File: unit_scaling/_modules.py (hunk 1 above)
Comment (thecharlieblake) [reply unit_scaling/_modules.py]: Fixed!

File: unit_scaling/_modules.py (hunk 4 above)
Comment (thecharlieblake) [reply unit_scaling/_modules.py]: Good point - thanks for spotting this. I've just disabled the `str` type for now as you suggest

**Pull request merged**